qiskit/_jobprocessor.py

```diff
@@ -17,109 +17,31 @@ def run_local_backend(qobj):
-        if isinstance(circuit['compiled_circuit_qasm'], bytes):
-            api_jobs.append({'qasm': circuit['compiled_circuit_qasm'].decode()})
-        else:
-            api_jobs.append({'qasm': circuit['compiled_circuit_qasm']})
-
-    seed0 = qobj['circuits'][0]['config']['seed']
-    output = api.run_job(api_jobs, qobj['config']['backend'],
-                         shots=qobj['config']['shots'],
-                         max_credits=qobj['config']['max_credits'],
-                         seed=seed0)
-    if 'error' in output:
-        raise ResultError(output['error'])
-
-    job_result = _wait_for_job(output['id'], api, wait=wait, timeout=timeout, silent=silent)
-    job_result['name'] = qobj['id']
-    job_result['backend'] = qobj['config']['backend']
-    this_result = Result(job_result, qobj)
-    return this_result
-
-def _wait_for_job(jobid, api, wait=5, timeout=60, silent=True):
-    """Wait until all online ran circuits of a qobj are 'COMPLETED'.
-
-    Args:
-        jobid:  is a list of id strings.
-        api (IBMQuantumExperience): IBMQuantumExperience API connection
-        wait (int):  is the time to wait between requests, in seconds
-        timeout (int):  is how long we wait before failing, in seconds
-        silent (bool): is an option to print out the running information or
-            not
-
+        wait (float): seconds between run attempts
+        timeout (float): seconds
     Returns:
-        A list of results that correspond to the jobids.
-
+        Result object.
     Raises:
-        QISKitError:
+        QISKitError: if "ERROR" string in server response.
     """
-    timer = 0
-    job_result = api.get_job(jobid)
-    if 'status' not in job_result:
-        from pprint import pformat
-        raise QISKitError("get_job didn't return status: %s" % (pformat(job_result)))
-
-    while job_result['status'] == 'RUNNING':
-        if timer >= timeout:
-            return {'status': 'ERROR', 'result': 'Time Out'}
-        time.sleep(wait)
-        timer += wait
-        if not silent:
-            print('status = %s (%d seconds)' % (job_result['status'], timer))
-        job_result = api.get_job(jobid)
-
-        if 'status' not in job_result:
-            from pprint import pformat
-            raise QISKitError("get_job didn't return status: %s" % (pformat(job_result)))
-        if (job_result['status'] == 'ERROR_CREATING_JOB' or
-                job_result['status'] == 'ERROR_RUNNING_JOB'):
-            return {'status': 'ERROR', 'result': job_result['status']}
-
-    # Get the results
-    job_result_return = []
-    for index in range(len(job_result['qasms'])):
-        job_result_return.append({'data': job_result['qasms'][index]['data'],
-                                  'status': job_result['qasms'][index]['status']})
-    return {'status': job_result['status'], 'result': job_result_return}
+    BackendClass = qiskit.backends.get_backend_class(qobj['config']['backend'])
```

Comment:
This is one of the simplifications I wanted to make with the QuantumJob I mentioned above. I think once the remote backends accept qobj we would then only need to have one run_backend() function. Actually at that point it might not even be necessary to have a separate function.